### ISSUE TITLE
CLI: extend function to complete pvc-copy leading to workload-copy

### ISSUE THREAD

**Issue by sallyom** [backube/scribe#148]
With scribe plugin, create a complete `pvc-copy` command that:
1) created replicationDestination
2) creates replicationSource
3) syncs ssh-secret
4) creates PVC in destination from copyMethod

Then, extend that to `workload-copy` where `workload` can be `deployment, statefulset, etc`

**Comment by sallyom** [comment]
As I see it,  `kubectl scribe replicate` or `kubectl scribe pvc-copy` or `kubectl scribe sync` (please reply with preference :smile:  ) will look something like:
```
kubectl scribe sync \
    --replication-source <repSourceName> \
    --replication-dest <repDestinationName>
``` 
where we extract src, dest values from the repSource, repDest but will also need to know contexts, namespaces for each, all of  which can be provided from a config file

*or*
```
kubctl scribe sync
``` 
where the replicationDestination and replicationSource will be created from set of options passed in from config file and/or command-line. 

So, if no replication-source passed, create it && if no replication-destination passed, create that, too. However if they are already existing and are provided, use them (and possible renew the pvc, if told to do so, and once that seamless re-sync is figured out).

**Comment by JohnStrunk** [comment]
I think we have a couple different use-cases mashed together right now. (at least that's how I read it)
1) The original notion of pvc-copy - Same cluster, same namespace, just moving from 1 StorageClass to another. I'd expect the user's intent would be that this setup is short-lived... sync the data, pause the app, final sync, tear down the replication and start the app on the new.
2) As a helper for setting up cross-cluster replication.

Maybe we can get a single (sub-)command to do both, but I'm not sure. There's also the ensuring replication followed by teardown/cleanup.

Let's pick one and go w/ it. For (1) i'd say something like `copy-pvc` but for (2) `replicate`.

**Comment by sallyom** [comment]
For 1. above, enable this w/ scribe plugin, correct? https://drive.google.com/file/d/1oKc6ACT4XIOtMqirsqi2gosgjQt9e-LA/view

**Comment by JohnStrunk** [comment]
Yes, minus the deployment scaling/editing. If the user says (at a minimum):
```
$ kubectl scribe copy-pvc --namespace myns --source pvc1 --destination pvc2 --storageclass sc2
```
we should be able to:
* Create the destination PVC of correct size and (new) SC
* Set up the replication (potentially defaulting to `copyMethod: none` on both src & dest), and leave off the schedule so it runs aggressively

We'd then need some sort of "finalize" operation that waits for a final complete replication and removes the RS/RD, leaving just the PVCs. (`kubectl scribe copy-pvc --finalize ...`?)

This allows the larger migration flow to be built on top by us or someone else inserting the scaling/editing (1) prior to the finalize and (2) after the teardown

**Comment by sallyom** [comment]
Thanks, I have a better understanding now, getting to work!

**Comment by sallyom** [comment]
plugin update has merged, as it stands now there's (for rsync):
```
$ kubectl scribe start-replication
$ kubectl scribe set-replication
$ kubectl scribe continue-replication
$ kubectl scribe remove-replication
```
I'm closing this, this issue is resolved.
